File: sdkwork-app-sdk-typescript/src/types/page-collection-vo.ts
import type { CollectionVO } from './collection-vo';
import type { PageableObject } from './pageable-object';
import type { SortObject } from './sort-object';

export interface PageCollectionVO {
  totalElements?: number;
  totalPages?: number;
  size?: number;
  content?: CollectionVO[];
  number?: number;
  first?: boolean;
  last?: boolean;
  numberOfElements?: number;
  sort?: SortObject;
  pageable?: PageableObject;
  empty?: boolean;
}

File: sdkwork-app-sdk-kotlin/src/main/kotlin/com/sdkwork/app/PageCollectionVO.kt
package com.sdkwork.app

data class PageCollectionVO(
    val totalElements: Int? = null,
    val totalPages: Int? = null,
    val size: Int? = null,
    val content: List<CollectionVO>? = null,
    val number: Int? = null,
    val first: Boolean? = null,
    val last: Boolean? = null,
    val numberOfElements: Int? = null,
    val sort: SortObject? = null,
    val pageable: PageableObject? = null,
    val empty: Boolean? = null
)
